src/silo/config/preprocessing_config.cpp

```diff
@@ -53,102 +159,80 @@ std::optional<std::filesystem::path> PreprocessingConfig::getNdjsonInputFilename
              : std::nullopt;
 }
 
-void PreprocessingConfig::overwrite(const silo::config::AbstractConfigSource& config) {
-   if (auto value = config.getString(INPUT_DIRECTORY_OPTION)) {
-      SPDLOG_DEBUG(
-         "Using {} as passed via {}: {}",
-         INPUT_DIRECTORY_OPTION.toString(),
-         config.configType(),
-         *value
-      );
-      input_directory = *value;
+void PreprocessingConfig::overwriteFrom(const VerifiedConfigAttributes& config_source) {
+   if (auto var = config_source.getPath(preprocessingConfigOptionKey())) {
+      preprocessing_config = var.value();
+   }
+   if (auto var = config_source.getPath(defaultPreprocessingConfigOptionKey())) {
+      default_preprocessing_config = var.value();
+   }
+   if (auto var = config_source.getPath(inputDirectoryOptionKey())) {
+      input_directory = var.value();
    }
-   if (auto value = config.getString(OUTPUT_DIRECTORY_OPTION)) {
-      SPDLOG_DEBUG(
-         "Using {} as passed via {}: {}",
-         OUTPUT_DIRECTORY_OPTION.toString(),
-         config.configType(),
-         *value
-      );
-      output_directory = *value;
+   if (auto var = config_source.getPath(outputDirectoryOptionKey())) {
+      output_directory = var.value();
    }
-   if (auto value = config.getString(INTERMEDIATE_RESULTS_DIRECTORY_OPTION)) {
-      SPDLOG_DEBUG(
-         "Using {} as passed via {}: {}",
-         INTERMEDIATE_RESULTS_DIRECTORY_OPTION.toString(),
-         config.configType(),
-         *value
-      );
-      intermediate_results_directory = *value;
+   if (auto var = config_source.getPath(intermediateResultsDirectoryOptionKey())) {
+      intermediate_results_directory = var.value();
    }
-   if (auto value = config.getString(PREPROCESSING_DATABASE_LOCATION_OPTION)) {
-      SPDLOG_DEBUG(
-         "Using {} as passed via {}: {}",
-         PREPROCESSING_DATABASE_LOCATION_OPTION.toString(),
-         config.configType(),
-         *value
-      );
-      preprocessing_database_location = *value;
+   if (auto var = config_source.getPath(preprocessingDatabaseLocationOptionKey())) {
+      preprocessing_database_location = var.value();
```

Comment:
This is Alex's code. But I would have included the .value() call, too, "if we have the value, assign the value to the field", is what we do here, whether the field is an optional or not is an independent matter, and it is not directly visible here whether it is. In Rust it would ask me to rewrap in a Some(), which I would accept to do, too. I think of them as optionals on a different level. Only if it is somehow guaranteed and part of a higher principle that is implicitly clear would I accept them as optionals on the same level. But really the kicker is that AFAIK it's not clear that the fields are optional and that it pays off to be super careful retrieving values from an optional.

Reply:
The fields are declared as optional in the `.h` file. Translated to Rust, this line would be `preprocessing_database_location = Some(var.unwarp());`, only that in C++, constructing an optional from a value is done implicitly.

Reply:
In Rust we would already have an

`if let Some var = getValue(..) {
   preprocessing_database_location = Some(var);
}`

I would also advocate that this would be the best version to have in Rust. As it is way less likely to introduce errors when refactoring. There was a great talk at rustlab.it on patterns like these:
https://rustlab.it/talks/fearless-refactoring-and-the-art-of-argument-free-rust

Reply:
Ok, the current version is at least consistent (the same for all fields). I think I would still prefer removing `.value()`, but I also don't have a really strong opinion here.